perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_attribute_def_virt_eduPersonScopedAffiliations.java

```diff
@@ -107,12 +107,61 @@ public Attribute getAttributeValue(PerunSessionImpl sess, User user, AttributeDe
 	}
 
 	@Override
-	public List<Pattern> getPatternsForMatch() {
-		List<Pattern> patterns = super.getPatternsForMatch();
-		patterns.add(userAllAttrsRemovedPattern);
-		patterns.add(userEPSAMARemovePattern);
-		patterns.add(userEPSAMASetPattern);
+	public List<AuditEvent> resolveVirtualAttributeValueChange(PerunSessionImpl perunSession, AuditEvent message) throws InternalErrorException, WrongReferenceAttributeValueException, AttributeNotExistsException, WrongAttributeAssignmentException {
+		List<AuditEvent> resolvingMessages = super.resolveVirtualAttributeValueChange(perunSession, message);
```

Comment:
If you implemet all of the logic for checking events, you do not need to call super here.